editoast/src/core/conflict_detection.rs

```diff
@@ -49,19 +61,29 @@ pub struct Conflict {
     pub conflict_type: ConflictType,
 }
 
-#[derive(Debug, Clone, Copy, Serialize, Deserialize, ToSchema)]
+#[derive(Debug, Clone, Deserialize, Serialize, ToSchema)]
+pub struct ConflictWithWorkSchedules {
+    #[serde(flatten)]
+    pub conflict: Conflict,
+    /// List of work schedule ids involved in the conflict
+    pub work_schedule_ids: Vec<i64>,
+}
```

Comment:
Well not really, we still can define a `Routing` with a non-empty `Vec` with is incorrect. Should we do something like the following?
```rust
pub enum ConflictType {
  Spacing(Vec<i64>),
  Routing,
}
```

Reply:
We discussed with @Erashin and TLDR, concluded that all solutions seems to bring considerable complexity so we'll keep a simple data structure that might be corrupted but hey 🤷. 

First of all, putting `work_schedule_ids` in the enum `ConflictType`, when `train_ids` involved in the conflict are already in `Conflict` doesn't make sense: both categories of ids should probably live in the same place. If we were to do something. That means we could go to something like this.

```rust
pub enum ConflictType {
  Spacing {
    train_ids: Vec<i64>,
    workschedule_ids: Vec<i64>,
  },
  Routing {
    train_ids: Vec<i64>,
  },
}
```
but that would mean a more important refactoring (pretty large and not in the scope of this PR).

On top of that, it's not entirely clear that OpenAPI would support this data structure, and even more, not even sure that codegen from `utoipa` to OpenAPI, then from OpenAPI to Typescript would work well. Also, this would probably means to manually implement the deserializer for converting what `core` is sending into this new Rust structure (and implementing manually the deserializer is no fun, it adds complexity).

Having said all that, we took the pragmatic solution to just move the `work_schedule_ids` next to `train_ids` in `Conflict`. That means that the data structure may contain corrupted data (i.e. it's possible to encode a `ConflictType::Routing` with some `work_schedule_ids`).